amd/src/courseformat/courseindex/section.js

```diff
@@ -93,20 +113,27 @@ export default class Component extends ComponentBase {
                 target = state.section.get(this.course.sectionlist[target.number - 1]);
             }
             const moveDirection = Math.sign(target.number - origin.number);
-            let targetEnd = target;
-            while (this.course.sectionlist.length > targetEnd.number + 1
-                    && state.section.get(this.course.sectionlist[targetEnd.number + 1]).levelsan > origin.levelsan) {
-                targetEnd = state.section.get(this.course.sectionlist[targetEnd.number + 1]);
+            let targetShow = target;
+            let targetShowBorder = moveDirection;
+            if (moveDirection > 0 && !target.indexcollapsed) {
+                let targetChild = target;
+                while (this.course.sectionlist.length > targetChild.number + 1
+                && state.section.get(this.course.sectionlist[targetChild.number + 1]).levelsan > target.levelsan) {
+                    targetChild = state.section.get(this.course.sectionlist[targetChild.number + 1]);
+                    if (targetChild.levelsan <= origin.levelsan) {
+                        targetShow = targetChild;
+                        targetShowBorder = -1;
+                        break;
+                    }
+                }
             }
             const targetHTML = document.querySelector(
-                ".courseindex-section[data-id='" + (moveDirection <= 0 ? target : targetEnd).id + "']");
+                ".courseindex-section[data-id='" + targetShow.id + "']");
             // The relative move of section depends on the section number.
-            if (moveDirection > 0) {
-                targetHTML.classList.remove(this.classes.DROPUP);
+            if (targetShowBorder > 0) {
                 targetHTML.classList.add(this.classes.DROPDOWN);
             } else {
```

Comment:
We need to ignore when a section is dragged over itself.
```suggestion
            } else  if (targetShowBorder < 0) {
```